content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2290.mdx

```diff
@@ -0,0 +1,194 @@
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+1.  Optionally introspect your **production database** and double-check that constraint names in production and development are in sync. If you have external applications or scripts that rely on a specific constraint name in production, make sure that these constraint names are represented in the Prisma schema.
+
+    In the following example, the `Post` model has a foreign key constraint with a custom name in production that does not match development:
+
+      <TabbedContent tabs={[<FileWithIcon text="Development environment" icon="prisma"/>, <FileWithIcon text="Production environment" icon="prisma"/>]}>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+      </tab>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Production_Name")
+    }
+    ```
+
+      </tab>
+
+      </TabbedContent>
+
+    To synchronize constraint names between development and production, make sure that the constraint name in your Prisma schema is `map: "Custom_Production_Name"`.
+
+1.  Optionally rename any explicitly defined constraint names where you do not want to use Prisma's naming convention, but do want to use a different constraint name:
+
+    ```prisma highlight=2;delete|3;add
+    model User {
+      id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+      id    Int    @id(map: "User_ID_Primary_Key") @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+    ```
+
+1.  Optionally _remove_ the `map` field from any constraint or index attribute where you want to use the Prisma default:
+
+    ```prisma highlight=5;delete|6;add
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+      author User? @relation(fields: [authorName], references: [name])
+    }
+    ```
+
+1.  Run the `prisma migrate dev` command to apply the changes to the database and generate a new migration:
+
+    ```terminal
+    npx prisma migrate dev
+    ```
+
+1.  Deploy the migration to production and test environments.
+
+    ```terminal
+    npx prisma migrate deploy
+    ```
+
+#### Upgrade path 2: I want to use Prismas default constraint and index names
```

Comment:
Isn't there also a upgrade path 3 "Do weird combination of things"?

Reply:
Maybe. You can of course mix and match.

Reply:
Optimally the document would be so clear, that people can read the intro and already understand the general problem and the options, then can follow some "guides" that spell out the exact tiny steps.